merlin/models/tf/core.py

```diff
@@ -1788,14 +1795,22 @@ def __init__(
         self.task_block = task_block
         self._task_name = task_name
         self.pre = pre
-        self.pre_metrics = pre_metrics
+        self.pre_eval_topk = pre_eval_topk
+
+        if metrics is not None:
+            if isinstance(metrics, SequenceCollection):
+                metrics = list(metrics)
+            else:
+                metrics = [metrics]
```

Comment:
Seems like this allows you to pass in a single metric, if that’s the case we should update the type-hint in the class.

Reply:
Created a new `MetricOrMetrics = Union[Sequence[MetricOrMetricClass], MetricOrMetricClass]` that is now used in tasks metrics arguments